docs/apache-airflow/core-concepts/dags.rst

```diff
@@ -484,6 +484,125 @@ You can also combine this with the :ref:`concepts:depends-on-past` functionality
+
+Setup "scope"
+"""""""""""""
+
+We require that a setup always have a teardown in order to have a well-defined scope. If you wish to only add a teardown task or only a setup task, you may use EmptyOperator as your "empty setup" or "empty teardown".
+
+The "scope" of a setup will be determined by where the teardown is.  Tasks between a setup and its teardown are in the "scope" of the setup / teardown pair. Example:
+
+.. code-block:: python
+
+    s1 >> w1 >> w2 >> t1.as_teardown(setups=s1) >> w3
+    w2 >> w4
+
+In the above example, w1 and w2 are "between" s1 and t1 and therefore are assumed to require s1. Thus if w1 or w2 is cleared, so too will be s1 and t1.  But if w3 or w4 is cleared, neither s1 nor t1 will be cleared.
+
+Controlling dag run state
+"""""""""""""""""""""""""
+
+Another feature of setup / teardown tasks is you can choose whether or not the teardown task should have an impact on dag run state.  Perhaps you don't care if the "cleanup" work performed by your teardown task fails, and you only consider the dag run a failure if the "work" tasks fail.  By default, teardown tasks are not considered for dag run state.
+
+Continuing with the example above, if you want the run's success to depend on ``delete_cluster``.  Then set property ``on_failure_fail_dagrun=True`` when setting ``delete_cluster`` as teardown:
+
+.. code-block:: python
+
+  create_cluster >> run_query >> delete_cluster.as_teardown(setups=create_cluster, on_failure_fail_dagrun=True)
+
+Authoring with task groups
+""""""""""""""""""""""""""
+
+When arrowing from task group to task group, or from task group to task, we ignore teardowns.  This allows teardowns to run in parallel, and allows dag execution to proceed even if teardown tasks fail.
+
+Consider this example:
+
+.. code-block:: python
+
+    with TaskGroup("my_group") as tg:
+        s1 = my_setup()
+        w1 = my_work()
+        t1 = my_teardown()
+        s1 >> w1 >> t1.as_teardown(setups=s1)
+    w2 = other_work()
+    tg >> w2
+
+If ``t1`` were not a teardown task, then this dag would effectively be ``s1 >> w1 >> t1 >> w2``.  But since we have marked ``t1`` as a teardown, it's ignored in ``tg >> w2``.  So the dag is equivalent to the following:
+
+.. code-block:: python
+
+    s1 >> w1 >> [t1.as_teardown(setups=s1), w2]
+
+Now let's consider an example with nesting:
+
+.. code-block:: python
+
+    with TaskGroup("my_group") as tg:
+        s1 = my_setup()
+        w1 = my_work()
+        t1 = my_teardown()
+        s1 >> w1 >> t1.as_teardown(setups=s1)
+    w2 = other_work()
+    tg >> w2
+    dag_s1 = dag_setup1()
+    dag_t1 = dag_teardown1()
+    dag_s1 >> [tg, w2] >> dag_t1.as_teardown(dag_s1)
+
+In this example s1 is downstream of dag_s1, so it must wait for dag_s1 to complete successfully.  But t1 and dag_t1 can run concurrently, because t1 is ignored in the expression ``tg >> dag_t1``.  If you clear w1, it will clear dag_s1 and dag_t1, but not anything in the task group.
```

Comment:
```suggestion
In this example ``s1`` is downstream of ``dag_s1``, so it must wait for ``dag_s1`` to complete successfully.  But ``t1`` and ``dag_t1`` can run concurrently, because ``t1`` is ignored in the expression ``tg >> dag_t1``.  If you clear ``w2``, it will clear ``dag_s1`` and ``dag_t1``, but not anything in the task group.
```

Reply:
I think you mean w2 will cause dag levels to be cleared, but nothing in the TG... since w1 is in the TG...

Reply:
yup, thanks